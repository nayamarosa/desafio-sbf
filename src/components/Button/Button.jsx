import React from 'react';

const Button = ({ id, text, onClick, ...props}) => {
  return (
    <button 
      id={id} 
      onChange={onClick} 
      {...props}
    >
      {text}
    </button>
  ) 
}

export default Button;