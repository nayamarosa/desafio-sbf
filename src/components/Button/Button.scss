.button {
  padding: 16px 24px;
  background-color: none;
  @include font(16px, 700, 1);
  border: none;
  border-radius: 50px;
  &__primary {
    background-color: $black;
    color: $white;
  }
}