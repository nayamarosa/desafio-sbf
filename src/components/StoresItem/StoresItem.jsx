import React from "react";

import Icon from "../Icon";

const StoresItem = ({ name, adress, onClick }) => {
  return (
    <li className="stores__list-item">
      <p className="stores__list-item--title">{name}</p>
      <a href="/" className="show-map" onClick={onClick}>
        <Icon icon='./images/icon_pin_link.svg' alt="Ver no mapa" />
        <p className="stores__list-item--map">Ver no mapa</p>
      </a>
      <div className="stores__list-item-group">
        <p className="stores__list-item--address">{adress}</p>
        <p className="stores__list-item--time">Atendimento:</p>
        <p className="stores__list-item--time">Segunda a Sábado 10h às 22h | Domingo 11h às 20h</p>
        <span>Disponível em 4 dias úteis</span>
      </div>
    </li>
  )
}

export default StoresItem;