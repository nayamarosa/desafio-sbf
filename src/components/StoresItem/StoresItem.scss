.stores__list {
  &-item {
    margin: 16px 0;
    padding: 0 16px;
    &--title {
      margin-bottom: 8px;
      @include font(24px, 500, auto);
    }
    &--map {
      margin-bottom: 16px;
      @include font(16px, 400, 1.5);
      text-decoration: underline;
    }
    &-group {
      padding: 24px;
      background-color: $white;
      color: $dark-gray;
      border-radius: 8px;
      & span {
        @include font(12px, 700, 1.5);
        color: #006611;
      }
    }
    &--address {
      margin-bottom: 8px;
      @include font(14px, 400, 1.5);
    }
    &--time {
      @include font(12px, 400, 1.5);
    }
  }
}