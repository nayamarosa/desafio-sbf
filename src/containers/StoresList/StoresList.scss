.stores {
  &__title {
    margin-bottom: 32px;
    @include font(32px, 500, auto);
    text-transform: capitalize;
  }
  &__form {
    width: 100%;
    & button {
      display: none;
    }
  }
  &__list {
    margin: 32px 0;
    padding: 16px 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    background-color: $light-gray;
    border-radius: 8px;
    &-filter {
      margin-bottom: 16px;
      padding: 0 16px;
      display: flex;
      justify-content: flex-end;
      align-items: center;
      & p {
        margin-right: 24px;
        @include font(14px, 400, 1.5);
      }
    }
  }
}