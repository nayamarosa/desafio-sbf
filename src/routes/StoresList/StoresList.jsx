import React from "react";

const StoresList = () => {
  return (
    <p>stores</p>
  );
}

export default StoresList;