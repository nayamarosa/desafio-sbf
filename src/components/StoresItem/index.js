import StoresItem from "./StoresItem";

export default StoresItem;