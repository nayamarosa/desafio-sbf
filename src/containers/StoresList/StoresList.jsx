import React, { useState, useEffect } from "react";
import Button from "../../components/Button/Button";

import api from '../../api'
import Input from '../../components/Input';
import Icon from '../../components/Icon'
import StoresItem from '../../components/StoresItem';

export const getCoordinates = (inputValue) => {
  return inputValue.split(',');
}

export const getDistance = (inputValue, api) => {
  return api.map((item) => {
  const latitude = parseFloat(item.latitude - getCoordinates(inputValue)[0]);
  const longitude = parseFloat(item.longitude - getCoordinates(inputValue)[1]);
  const distanceCalc = Math.sqrt(Math.pow(latitude, 2) + Math.pow(longitude, 2))
    console.log(distanceCalc)
    return {...item, distance: distanceCalc}
  })
}

const StoresList = () => {
  const [apiData, setApiData] = useState([]);
  const [inputSearch, setInputSearch] = useState('');
  const [closestStores, setClosestStores] = useState([]);

  const apiImport = async() => {
    setApiData(await api.apiRes());
  }

  useEffect(() => {
    apiImport();
    console.log(closestStores)
  }, [])

  const search = (e, inputSearch) => {
    e.preventDefault();
    setClosestStores(getDistance(inputSearch, apiData).sort((a, b) => a.distance - b.distance).slice(0, 3));
  }

  return (
    <>
      <h1 class="stores__title">lojas</h1>
      <form class="stores__form">
        <Input
          placeholder="Busque por endereço ou CEP"
          type="text"
          data-testid="input-value"
          onChange={(e) => setInputSearch(e.target.value)}
        />
        <Button
          text="Buscar"
          type="submit"
          data-testid="search-button"
          onClick={(e) => search(e, inputSearch)}
        />
      </form>
      <>
      { closestStores.length > 0 &&
        <section class="stores__result">
          <div class="stores__result-filter">
            <p>Menor distância</p>
            <Icon icon="./images/icon_seta_baixo.svg" alt="Seta para  baixo"/>
          </div>
          <ul class="stores__result-list" data-testid="list">
            {closestStores.map((item, index) => <StoresItem name={item.name} adress={item.adress} key={index}/>)}
          </ul>
        </section>
      }
      </>
    </>
  );
}

export default StoresList;