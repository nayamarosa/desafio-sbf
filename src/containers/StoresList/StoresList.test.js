import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import StoresList, { getCoordinates, calcDistance, getDistance } from './StoresList'
import StoresItem from "../../components/StoresItem";
import Button from "../../components/Button/Button";

describe('Stores List Route', () => {
  it('coordinates in array', () => {
    const getCoor = '-23.605033, -46.576552';
    const coordinates = getCoordinates(getCoor);
    expect(coordinates).toStrictEqual(['-23.605033', ' -46.576552'])
  })

  it('same snapshot list container', () => {
    const storesList = [
      { name: 'loja1', adress: 'rua1' },
      { name: 'loja2', adress: 'rua2' },
      { name: 'loja3', adress: 'rua3' }
    ]

    const { container } = render(
      storesList.map((item, index) => <StoresItem name={item.name} adress={item.adress} key={index}/>)
    )
    expect(container).toMatchSnapshot();
  })

  it('calc ok', () => {
    const getCoor = '-23.605033, -46.576552';
    const getObj = {
      "name": "NFS Alexania",
      "number": 6322,
      "latitude": "-16.121590171053200",
      "longitude": "-48.39803935435456",
      "adress": "BR 060- km22 – Zona Rural"
    }
    const calculate = calcDistance(getObj, getCoor);
    expect(calculate).toStrictEqual(7.701930456462772)
  })

  it('distance add to object', () => {
    const getCoor = '-23.605033, -46.576552';
    const getObj = [{
      "name": "NFS Alexania",
      "number": 6322,
      "latitude": "-16.121590171053200",
      "longitude": "-48.39803935435456",
      "adress": "BR 060- km22 – Zona Rural"
    }]
    const distance = getDistance(getCoor, getObj);
    expect(distance).toStrictEqual([{
      "name": "NFS Alexania",
      "number": 6322,
      "latitude": "-16.121590171053200",
      "longitude": "-48.39803935435456",
      "adress": "BR 060- km22 – Zona Rural",
      "distance": 7.701930456462772
    }])
  })
}) 