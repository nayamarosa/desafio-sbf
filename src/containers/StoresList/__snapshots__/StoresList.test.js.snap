// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stores List Route same snapshot list container 1`] = `
<div>
  <li
    class="stores__list-item"
  >
    <p
      class="stores__list-item--title"
    >
      loja1
    </p>
    <a
      class="show-map"
      href="/"
    >
      <img
        alt="Ver no mapa"
        src="./images/icon_pin_link.svg"
      />
      <p
        class="stores__list-item--map"
      >
        Ver no mapa
      </p>
    </a>
    <div
      class="stores__list-item-group"
    >
      <p
        class="stores__list-item--address"
      >
        rua1
      </p>
      <p
        class="stores__list-item--time"
      >
        Atendimento:
      </p>
      <p
        class="stores__list-item--time"
      >
        Segunda a Sábado 10h às 22h | Domingo 11h às 20h
      </p>
      <span>
        Disponível em 4 dias úteis
      </span>
    </div>
  </li>
  <li
    class="stores__list-item"
  >
    <p
      class="stores__list-item--title"
    >
      loja2
    </p>
    <a
      class="show-map"
      href="/"
    >
      <img
        alt="Ver no mapa"
        src="./images/icon_pin_link.svg"
      />
      <p
        class="stores__list-item--map"
      >
        Ver no mapa
      </p>
    </a>
    <div
      class="stores__list-item-group"
    >
      <p
        class="stores__list-item--address"
      >
        rua2
      </p>
      <p
        class="stores__list-item--time"
      >
        Atendimento:
      </p>
      <p
        class="stores__list-item--time"
      >
        Segunda a Sábado 10h às 22h | Domingo 11h às 20h
      </p>
      <span>
        Disponível em 4 dias úteis
      </span>
    </div>
  </li>
  <li
    class="stores__list-item"
  >
    <p
      class="stores__list-item--title"
    >
      loja3
    </p>
    <a
      class="show-map"
      href="/"
    >
      <img
        alt="Ver no mapa"
        src="./images/icon_pin_link.svg"
      />
      <p
        class="stores__list-item--map"
      >
        Ver no mapa
      </p>
    </a>
    <div
      class="stores__list-item-group"
    >
      <p
        class="stores__list-item--address"
      >
        rua3
      </p>
      <p
        class="stores__list-item--time"
      >
        Atendimento:
      </p>
      <p
        class="stores__list-item--time"
      >
        Segunda a Sábado 10h às 22h | Domingo 11h às 20h
      </p>
      <span>
        Disponível em 4 dias úteis
      </span>
    </div>
  </li>
</div>
`;
